R/Set_up_subfunctions.R

```diff
@@ -0,0 +1,338 @@
+  #=======================================#
+  # preparing age_width & age_rate params #
+  
+  # age_width calc: Age-structured model #
+  if (number_age_classes_pig > 1) {
+    for (i in number_age_classes_pig) {
+      # width of 1 month for each age class (symmetrical age classes)
+      age_width_pig[1:number_age_classes_pig] <- pig_age_class_width
+      }
+    }
+  
+  # create vector of length n age classes for ODIN (& create dimensions for other variables)
+  if (number_age_classes_pig == 1) {
+    na_pig <- 1
+    }
+  
+  if (number_age_classes_pig > 1) {
+    na_pig <- as.integer(length(age_width_pig)) 
+    }
+  
+  # calculate age rate (function of age width) for non age-strcutured model # 
+  if (number_age_classes_pig == 1) {
+    age_rate_pig <- 0
+    }
+  
+  # calculate age rate (function of age width) for age-strcutured model # 
+  if (number_age_classes_pig > 1) {
+    age_rate_pig <- c()
+    
+    for (i in na_pig) {
+      # age rate leaving age classes (to next) excluding last age class (e.g. 1:2)
+      age_rate_pig[1:(na_pig - 1)] <- 1 / age_width_pig[1:i - 1]
+      # no age rate leaving last age class
+      age_rate_pig[na_pig] <- 0
+      }
+    }
+  
+  return(list(age_width_pig, na_pig, age_rate_pig))
+}
+
+
+#' @title
+#' Calculate life table for pig population
+#' @description
+#' To set up population distribution at baseline: LIFE TABLES APPROACH for Age-structured model 
+#' Based on code from Verity et al: https://github.com/mrc-ide/covfefe 
+#' Further reference here FAO paper: http://www.fao.org/tempref/docrep/fao/008/a0212e/a0212E09.pdf  
+#' 
+#' @param number_age_classes the number of age classes (pigs)
+#' @param slgt_age_min the age from which pigs are slaughtered (months), user-specified
+#' @param slgtage age-class where slaughter rate first applied to
+#' @param slgtage_bfr age-class before slaughter age first begins
+#' @param dP pig death rate (natural)
+#' @param dPslg pig slaughter-age death rate
+#' @param na_pig number age classes for pigs (processed)
+#'
+#' @return life table output
+#' @export
+life_tables_pigs_func <- function(number_age_classes, slgt_age_min, slgtage, slgtage_bfr,
+                                  dP = dP, dPslg = dPslg, na_pig){
+
+  number_age_classes_pig <- number_age_classes
+  slaughter_age_min <- slgt_age_min
+  slgtage <- slgtage
+  slgtage_bfr <- slgtage_bfr
+  
+#==============================================================================================#
+# To set up population distribution at baseline: LIFE TABLES APPROACH for Age-structured model #
+#                                         #
+# FAO paper: http://www.fao.org/tempref/docrep/fao/008/a0212e/a0212E09.pdf                     #        
+#
+
+# non-age structured 
+# if (slaughter_age_min == 0) {
+#   age1toslg <- 0
+#   ageslgtoN <- length(c(1:na_pig))
+# }
+
+  if (number_age_classes_pig == 1){ 
+    stop("error: do not need life tables approach for 1 age class (non-age strcutured model")
+  }
```

Comment:
Do we ever get here? If the user puts `number_age_classes_pig = 1` why does this function get called?